packages/logging-controller/src/LoggingController.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  BaseController,
+  BaseConfig,
+  BaseState,
+} from '@metamask/base-controller';
+import { v1 as random } from 'uuid';
+
+export enum LogType {
+  eth_sign_proposed = 'eth_sign_proposed',
+  eth_sign_signed = 'eth_sign_signed',
+  eth_sign_rejected = 'eth_sign_rejected',
+  personal_sign_proposed = 'personal_sign_proposed',
+  personal_sign_signed = 'personal_sign_signed',
+  personal_sign_rejected = 'personal_sign_rejected',
+  eth_signTypedData_proposed = 'eth_signTypedData_proposed',
+  eth_signTypedData_signed = 'eth_signTypedData_signed',
+  eth_signTypedData_rejected = 'eth_signTypedData_rejected',
+  eth_signTypedData_v3_proposed = 'eth_signTypedData_v3',
+  eth_signTypedData_v3_signed = 'eth_signTypedData_v3',
+  eth_signTypedData_v3_rejected = 'eth_signTypedData_v3',
+  eth_signTypedData_v4_proposed = 'eth_signTypedData_v4_proposed',
+  eth_signTypedData_v4_signed = 'eth_signTypedData_v4_signed',
+  eth_signTypedData_v4_rejected = 'eth_signTypedData_v4_rejected',
+}
+
+/**
+ * @type Log
+ *
+ * Log representation
+ * @property id - Generated UUID associated with this log
+ * @property timestamp - Timestamp associated with this log
+ * @property type - Type of signature request
+ * @property data - Additional data about signature request
+ */
```

Comment:
Note: `@type` is used to denote the type of a variable or fragment of code, but the proper tag for type _definitions_ is `@typedef`. That said, we use TypeDoc to generate docs and it doesn't really honor either `@type` or `@typedef`, as we can just use types. The format here is the same as for a class, function or method:

```
Description of thing

Tag1
Tag2
Tag3
...
```

So this can be:

``` suggestion
/**
 * Log representation
 * 
 * @property id - Generated UUID associated with this log
 * @property timestamp - Timestamp associated with this log
 * @property type - Type of signature request
 * @property data - Additional data about signature request
 */

Reply:
Most of the jsdocs should be updated. WOuld appreciate a double check